pkg/tsdb/mysql/mysql.go

```diff
@@ -134,6 +134,24 @@ func (s *Service) getDataSourceHandler(pluginCtx backend.PluginContext) (*sqleng
 	return instance, nil
 }
 
+// CheckHealth pings the connected SQL database
+func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	dsHandler, err := s.getDataSourceHandler(req.PluginContext)
+	if err != nil {
+		return nil, err
+	}
+
+	err = dsHandler.Ping()
+
+	var driverErr *mysql.MySQLError
+	if errors.As(err, &driverErr) {
+		logger.Error("Check health failed", "error", err)
+		return &backend.CheckHealthResult{Status: backend.HealthStatusError, Message: driverErr.Message}, nil
+	}
```

Comment:
You should also handle `err != nil` when it's not `As mysql.MySQLError`:

```suggestion
	if err != nil {
		logger.Error("Check health failed", "error", err)
		var driverErr *mysql.MySQLError
		if errors.As(err, &driverErr) {
			return &backend.CheckHealthResult{Status: backend.HealthStatusError, Message: driverErr.Message}, nil
		}
		return &backend.CheckHealthResult{Status: backend.HealthStatusError, Message: err.Error()}, nil
	}
```

:point_up:  Here in my suggestion also consider `TransformQueryError` instead of `err.Error()` as suggested by @marefr